src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -110,18 +108,6 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
 
     <hr/>
 
-<%--    <% if (request.getSession().getAttribute("user") != null) { %>
-    <form action="/chat/<%= conversation.getTitle() %>" method="POST">
-        <input type="text" name="message">
-        <br/>
-        <button type="submit">Send</button>
-    </form>
-    <% } else { %>
-      <p><a href="/login">Login</a> to send a message.</p>
-    <% } %>
-
-    <hr/>
---%>
   </div>
 
 </body>
```

Comment:
This whole pull request looks decent to me so far. I really like how you write the abstract and inheritance classes. They look very organized.